test/model/test_driver.py

```diff
@@ -14,6 +14,7 @@ def test_from_state():
     for msg in ws_msgs[1:]:
         msg = json.loads(msg)
         assert msg["type"] == "event"
-        driver.receive_event(msg["event"])
+        driver_event = driver_pkg.DriverEvent(type=msg["event"], data=msg)
+        driver.receive_event(driver_event)
```

Comment:
There seems to be a missing key in the fixture data we use here. We get a `KeyError: 'source'`.

Reply:
Nevermind. It was just me.

Reply:
> I've now rebased on the updated model in master branch and applied this PR suggestion to that. I haven't built out the node and value protocols yet. I'll do that if we want to go ahead with this.

Looks good!. I could really use this merged in for the integration as I can reference the enums etc.